File: frontend/src/app/models/ligneDemandeRetour.ts
import { CommandLigne } from "./CommandLigne";
import { Article } from "./article";
import { DemandeRetour } from "./demandeRetour";
import { Motif } from "./motif";

export class LigneDemandeRetour{
    id:number
nbArticleRetenue:string
motif:Motif
motifId:number
article:Article
articleId:number
artcileIntitule:string
articleCode:string
nbArticleTotale:number
demandeRetour:DemandeRetour
demandeRetourId:number
commandeLigne:CommandLigne
commandeLigneId:number
constructor(id:number,nbArticleRetenue:string,motif:Motif,motifId:number
    ,article:Article,articleId:number,artcileIntitule:string,articleCode:string
    ,nbArticleTotale:number,demandeRetour:DemandeRetour,demandeRetourId:number
    ,commandeLigne:CommandLigne,commandeLigneId:number){
        this.id=id
        this.nbArticleRetenue=nbArticleRetenue
        this.motif=motif
        this.motifId=motifId
        this.article=article
        this.articleId=articleId
        this.artcileIntitule=artcileIntitule
        this.articleCode=articleCode
        this.nbArticleTotale=nbArticleTotale
        this.demandeRetour=demandeRetour
        this.demandeRetourId=demandeRetourId
        this.commandeLigne=commandeLigne
        this.commandeLigneId=commandeLigneId
}
}

File: backend/B2BApplication/Models/LigneDemandeRetour.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace B2BApplication.Models
{
    public class LigneDemandeRetour
    {
        [Key]
        public int id { get; set; }
        public int nbArticleRetenue { get; set; }
        public Motif motif { get; set; }
        [ForeignKey("motif")]
        public int motifId { get; set; }
        public Article article { get; set; }
        [ForeignKey("article")]
        public int articleId { get; set; }
        public string artcileIntitule { get; set; }
        public string articleCode { get; set; }
        public int nbArticleTotale { get; set; }
        public DemandeRetour demandeRetour { get; set; }
        [ForeignKey("demandeRetour")]
        public int demandeRetourId { get; set; }
        public CommandeLigne commandeLigne { get; set; }
        [ForeignKey("commandeLigne")]
        public int commandeLigneId { get; set; }
    }
}
